url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/compiler/inst/XInstSwitch.java

```diff
@@ -0,0 +1,276 @@
+package xscript.compiler.inst;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import xscript.XOpcode;
+import xscript.compiler.XCodeGen;
+import xscript.compiler.XDataOutput;
+import xscript.compiler.XJumpTarget;
+
+public class XInstSwitch extends XInst {
+
+	public XJumpTarget _default;
+	private HashMap<Object, XJumpTarget> switches = new HashMap<Object, XJumpTarget>();
+	private int resolvedDefault;
+	private long min=Long.MAX_VALUE;
+	private long max=Long.MIN_VALUE;
+	private HashMap<Object, Integer> resolvedSwitches = new HashMap<Object, Integer>();
+	private int resolvedDefault2;
+	private HashMap<Object, Integer> resolvedSwitches2 = new HashMap<Object, Integer>();
+	
+	public XInstSwitch(int line) {
+		super(line, XOpcode.SWITCH);
+	}
+
+	@Override
+	public boolean pointingTo(XInst inst) {
+		if(_default.is(inst))
+			return true;
+		for(XJumpTarget o:switches.values()){
+			if(o.is(inst))
+				return true;
+		}
+		return false;
+	}
+
+	@Override
+	public void resolve(XCodeGen codeGen, List<XInst> instructions) {
+		resolvedDefault = _default.resolve(codeGen);
+		resolvedDefault2 = _default.resolve(instructions);
+		for(Entry<Object, XJumpTarget> e:switches.entrySet()){
+			Object k = e.getKey();
+			resolvedSwitches.put(k, e.getValue().resolve(codeGen));
+			resolvedSwitches2.put(k, e.getValue().resolve(instructions));
+			if(k instanceof Number){
+				long l = ((Number)k).longValue();
+				if(min>l)
+					min=l;
+				if(max<l)
+					max=l;
+			}
+		}
+	}
+
+	@Override
+	public void replace(XCodeGen codeGen, XInst instruction, XInst with, List<XInst> instructions) {
+		_default.replace(instruction, with);
+		for(Entry<Object, XJumpTarget> e:switches.entrySet()){
+			e.getValue().replace(instruction, with);
+		}
+	}
+
+	@Override
+	public boolean isNormalJump() {
+		return false;
+	}
+
+	@Override
+	public boolean isAlwaysJump() {
+		return true;
+	}
+
+	@Override
+	public int tryWay(XCodeGen codeGen, int programPointer, int stackSize, int[] sizes) {
+		for(Integer i:resolvedSwitches2.values()){
+			codeGen.tryWay(i, stackSize, sizes);
+		}
+		return resolvedDefault2;
+	}
+	
+	@Override
+	public void toCode(XDataOutput dataOutput) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 30. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 89-197 into a separate method.